app/stac_api/models.py

```diff
@@ -361,27 +806,94 @@ def save(self, *args, **kwargs):  # pylint: disable=signature-differs
+                # updated
+                self.collection.update_temporal_extent(
+                    'up',
+                    self.__original_properties_datetime,
+                    self.properties_datetime,
+                    self.__original_properties_datetime,
+                    self.properties_datetime,
+                    self.pk
+                )
+            else:
+                # This is the case, when the item was defined by a start_ and
+                # end_datetime before and has been changed to only have a single
+                # datetime property. In that case, we hand over the old
+                # start_ and end_datetime values to the update function, so
+                # that a loop over all items will only be done, if really
+                # necessary.
+
+                self.collection.update_temporal_extent(
+                    'up',
+                    self.__original_properties_start_datetime,
+                    self.properties_datetime,
+                    self.__original_properties_end_datetime,
+                    self.properties_datetime,
+                    self.pk
+                )
+        elif self.__original_properties_start_datetime is not None and \
+            self.__original_properties_end_datetime is not None:
+            # This is the case, if an items values for start_ and/or end_datetime
+            # were updated.
             self.collection.update_temporal_extent(
-                self.properties_datetime, self.properties_datetime
+                'up',
+                self.__original_properties_start_datetime,
+                self.properties_start_datetime,
+                self.__original_properties_end_datetime,
+                self.properties_end_datetime,
+                self.pk
             )
         else:
+            # This is the case, when an item was defined by a single datetime
+            # before and has been changed to contain a start_ and an
+            # end_datetime value now.
             self.collection.update_temporal_extent(
-                self.properties_start_datetime, self.properties_end_datetime
+                'up',
+                self.__original_properties_datetime,
+                self.properties_start_datetime,
+                self.__original_properties_datetime,
+                self.properties_end_datetime,
+                self.pk
             )
+
         # adding a new item means update the bbox of the collection
         if self.pk is None:
             self.collection.update_bbox_extent('insert', self.geometry, self.pk, self.item_name)
         # update the bbox of the collection only when the geometry of the item has changed
         elif self.geometry != self._original_geometry:
             self.collection.update_bbox_extent('update', self.geometry, self.pk, self.item_name)
-        super().save(*args, **kwargs)
+        super(Item, self).save(*args, **kwargs)
```

Comment:
Why changing the way we call super here ? Even this is the same, this is dangerous if we move this function in a subclass of Item, then it will still call the `Model.save()` and not the `Item.save()` method. 
`super().method()` should always be preferred to `super(subclass, self).method()`

Reply:
Good point. I guess this was because I was following this example here (which was a good catch from @rebert):
https://stackoverflow.com/questions/1355150/when-saving-how-can-you-check-if-a-field-has-changed
You are right, this might cause unexpected behavior, I'll change it today when working on all the comments :+1:

Reply:
done